lib/tasks/pipeline_monitor.rake

```diff
@@ -70,12 +80,172 @@ class CheckPipelineRuns
+    # Tags added here are capitalized.  If a COMMENT tag already exists in metadata, prepend to it.
+    existing_comment = metadata['COMMENT']
+    comment_separator = "\n"
+    comment_separator = "" if existing_comment.blank?
+    comment_separator = "" if bm_comment.blank?
+    new_metadata = {}
+    # HACK: We want COMMENT and ORIGIN to appear at the top in the json dump, and this trick does it.
+    new_metadata['COMMENT'] = (bm_comment || "") + comment_separator + (existing_comment || "")
+    new_metadata['ORIGIN'] = s3path
+    metadata.each do |k, v|
+      next if k == "COMMENT"
+      next if k == "ORIGIN"
+      new_metadata[k] = v
+    end
+    known_organisms = metadata['verified_contents'].pluck('genome').join(", ")
+    bm_sample_params = {
+      name: bm_sample_name,
+      host_genome_id: bm_host.id,
+      project_id: bm_project.id,
+      user_id: bm_user.id,
+      input_files_attributes: input_files_attributes,
+      sample_organism: known_organisms,
+      web_commit: web_commit,
+      pipeline_commit: pipeline_commit,
+      pipeline_branch: bm_pipeline_branch,
+      sample_notes: JSON.pretty_generate(new_metadata)
+    }
+    @bm_sample = Sample.new(bm_sample_params)
+    # HACK: Not really sure why we have to manually set the status to STATUS_CREATED here,
+    # but if we don't, the sample is never picked up by the uploader.
+    @bm_sample.status ||= Sample::STATUS_CREATED
+    unless @bm_sample.save
+      raise "Error creating benchmark sample with #{JSON.pretty_generate(bm_sample_params)}."
+    end
+    Rails.logger.info("Benchmark sample #{@bm_sample.id} created successfully.")
+  end
+
+  def self.prop_get(dict, property, defaults)
+    return dict[property] if dict.key?(property)
+    defaults[property]
+  end
+
+  def self.benchmark_update(t_now)
+    Rails.logger.info("Benchmark update.")
+    config = JSON.parse(`aws s3 cp #{IDSEQ_BENCH_CONFIG} -`)
+    defaults = config['defaults']
+    web_commit = ENV['GIT_VERSION'] || ""
+    config['active_benchmarks'].each do |s3path, bm_props|
+      bm_environments = prop_get(bm_props, 'environments', defaults)
+      unless bm_environments.include?(Rails.env)
+        Rails.logger.info("Benchmark does not apply to #{Rails.env} environment: #{s3path}")
+        next
+      end
+      bm_project_name = prop_get(bm_props, 'project_name', defaults)
+      bm_proj = Project.find_by(name: bm_project_name)
+      unless bm_proj
+        Rails.logger.info("Benchmark requires non-existent project #{bm_project_name}: #{s3path}")
+        next
+      end
+      bm_frequency_hours = prop_get(bm_props, 'frequency_hours', defaults)
+      bm_frequency_seconds = bm_frequency_hours * 3600
+      unless bm_frequency_hours >= IDSEQ_BENCH_MIN_FREQUENCY_HOURS
+        Rails.logger.info("Benchmark frequency under #{IDSEQ_BENCH_MIN_FREQUENCY_HOURS} hour: #{s3path}")
+        next
+      end
+      bm_name_prefix = benchmark_sample_name_prefix(s3path)
+      sql_query = "
```

Comment:
Why not use active record given that is a single table query?

Reply:
Probably a good idea!